src/Worker/Core/Shims/TaskOrchestrationShim.cs

```diff
@@ -14,44 +15,38 @@ namespace Microsoft.DurableTask;
 /// </remarks>
 partial class TaskOrchestrationShim : TaskOrchestration
 {
-    readonly TaskName name;
     readonly ITaskOrchestrator implementation;
-    readonly WorkerContext workerContext;
-    readonly OrchestrationRuntimeState runtimeState;
-
+    readonly OrchestrationInvocationContext invocationContext;
     TaskOrchestrationContextWrapper? wrapperContext;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="TaskOrchestrationShim"/> class.
     /// </summary>
-    /// <param name="context">Context from the worker to make available to the orchestration runtime.</param>
-    /// <param name="name">The name of the orchestration.</param>
+    /// <param name="invocationContext">The invocation context for this orchestration.</param>
     /// <param name="implementation">The orchestration's implementation.</param>
     public TaskOrchestrationShim(
-        OrchestrationInvocationContext context,
-        TaskName name,
+        OrchestrationInvocationContext invocationContext,
         ITaskOrchestrator implementation)
     {
-        this.workerContext = context.WorkerContext;
-        this.runtimeState = context.RuntimeState;
-        this.name = name;
+        this.invocationContext = invocationContext;
         this.implementation = implementation;
     }
 
     /// <inheritdoc/>
     public override async Task<string?> Execute(OrchestrationContext innerContext, string rawInput)
     {
-        JsonDataConverterShim converterShim = new(this.workerContext.DataConverter);
+        JsonDataConverterShim converterShim = new(this.invocationContext.DataConverter);
         innerContext.MessageDataConverter = converterShim;
         innerContext.ErrorDataConverter = converterShim;
 
-        object? input = this.workerContext.DataConverter.Deserialize(rawInput, this.implementation.InputType);
+        object? input = this.invocationContext.DataConverter.Deserialize(rawInput, this.implementation.InputType);
 
-        this.wrapperContext = new(innerContext, this.name, this.workerContext, this.runtimeState, input);
+        ILogger contextLogger = this.invocationContext.LoggerFactory.CreateLogger("Microsoft.DurableTask");
```

Comment:
Previously we encapsulated the creation of loggers inside `SdkUtils.GetLogger(loggerFactory)`. Is there a reason to not do that here?

Reply:
Don't have `SdkUtils` here. I also plan on refactoring that later. We should not be using a single category, but rather multiple categories. We just have to make sure they all start with `Microsoft.DurableTask`. So this one I actually want to change to `.CreateLogger<TaskOrchestrationContext>()` (in another PR).